assets/src/application/ui/input/Button/ButtonGroup/index.tsx

```diff
@@ -1,16 +1,36 @@
-import React from 'react';
+import React, { cloneElement } from 'react';
 import classNames from 'classnames';
 
-import { ButtonGroup as ButtonGroupAdapter, ButtonGroupProps } from '@infraUI/inputs';
+import { ButtonGroup as ButtonGroupAdapter } from '@infraUI/inputs';
+import { ButtonGroupProps } from './types';
+import { cleanChildren } from './utils';
 
 import './style.scss';
 
-const ButtonGroup: React.FC<ButtonGroupProps> = ({ children, ...props }) => {
+const ButtonGroup: React.FC<ButtonGroupProps> = ({ buttonSize, children, ...props }) => {
 	const className = classNames(props.className, 'ee-btn-group');
+	const validChildren = cleanChildren(children);
+	const clones = validChildren.map((child, index) => {
+		const isFirst = index === 0;
+		const isLast = index === validChildren.length - 1;
+
+		// @ts-ignore
+		return cloneElement(child, {
+			// @ts-ignore
+			buttonSize: buttonSize || child.props.buttonSize,
+
+			// @ts-ignore
+			...(isFirst && { className: classNames(child.props.className, 'ee-btn--first') }),
+			// @ts-ignore
+			...(isLast && { className: classNames(child.props.className, 'ee-btn--last') }),
+			// @ts-ignore
+			...(!isFirst && !isLast && { className: classNames(child.props.className, 'ee-btn--no-border-radius') }),
+		});
+	});
```

Comment:
cloning is completely unnecessary imho, this can all be achieved using css pseudo selectors